src/python/dxpy/scripts/dx.py

```diff
@@ -3785,6 +3785,177 @@ def publish(args):
+                    # find a filename
+                    else:
+                        try: 
+                            file_results = list(dxpy.find_data_objects(classname="file", name=entity_name,project=target_project,folder=folderpath,describe=True,))
+                        except Exception as e:
+                            warn("Input '{}' is not valid: {}".format(":".join(path), e))
+                            continue
+                        
+                        if file_results and not args.all:
+                            picked_file = pick([ "{} ({})".format(result['describe']['name'], result['id']) for result in file_results],allow_mult=True)
+                            if picked_file == "*" :
+                                target_files+=[file['id'] for file in file_results]
+                            else:
+                                target_files.append(file_results[picked_file]['id'])
+                        else: 
+                            target_files+=[file['id'] for file in file_results]
+                
+                # once find a folder, skip the rest of inputs
+                if target_folder:
+                    break
+                # reset target project if all previous file paths failed
+                # otherwise, will keep using the target project id if any fileID is given, or filename is resolved
+                if not target_files:
+                    target_project = None
+                
+        return target_files, target_folder, target_project
+    
+    # resolve paths  
+    target_files, target_folder, target_project = get_archival_paths(args)
+
+    # check input: project ID + either files or one folder should be specified                               
+    request_input = get_valid_archival_input(args, target_files, target_folder, target_project)
+    
+    # set request mode: archive / unarchive
+    # set request command and add additional options
+    mode = args.request_mode    
+    archival_options = {}
+    if mode == "archival":
+        archival_options = {"allCopies": args.all_copies}
+        request_func = dxpy.api.project_archive
+    elif mode == "unarchival":
+        archival_options = {"rate": args.rate}
+        request_func = dxpy.api.project_unarchive        
+    request_input.update(archival_options)         
+    
+    # ask for confirmation if needed
+    if args.confirm and INTERACTIVE_CLI:
+        print("Will send the request:")       
+        print("{:>10}: {}".format("project",target_project))
+        confirm_msg="\n".join(["{:>10}: {}".format(k,v) for k,v in request_input.items()])
+        print(confirm_msg)
+        
+        if mode == "archival":
+            counts = len(set(target_files)) or len(list(dxpy.find_data_objects(project=target_project,folder=target_folder,classname="file",recurse=args.recurse)))
+            print('Will request in total {} file(s) for archival in {}'.format(counts,target_project))
```

Comment:
```suggestion
            print('Will tag {} file(s) for archival in {}'.format(counts,target_project))
```

Reply:
In case of a folder with a large number of files, can we just get the count of files?